src/fileformat/utils/other.cpp

```diff
@@ -99,5 +101,38 @@ std::vector<std::string> getSupportedArchitectures()
 	return result;
 }
 
+/**
+ * Compute entropy of given data
+ * @param data Data to compute entropy from
+ * @param dataLen Length of @a data
+ * @return entropy in <0,1>
+ */
+double computeDataEntropy(const std::uint8_t *data, std::size_t dataLen)
+{
+	std::array<std::size_t, 256> histogram{};
+	double entropy = 0;
+
+	if (!data)
+	{
+		return 0;
+	}
+
+	for (std::size_t i = 0; i < dataLen; i++)
+	{
+		histogram[data[i]]++;
+	}
+
+	for (auto frequency : histogram)
+	{
+		if (frequency)
+		{
+			double probability = static_cast<double>(frequency) / dataLen;
+			entropy -= probability * (log(probability) / log(2));
```

Comment:
This can be calculated using only single logarithm because `log_a(x) / log_a(b) = log_b(x)` so you can actually use logarithm of base-2 here. Also please use `std::` namespace instead of the functions in the global namespace.